src/xdocs/writingchecks.xml

```diff
@@ -660,15 +660,12 @@ public class MethodLimitCheck extends AbstractCheck
           <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.5">
             tokens (identifiers, keywords)</a>
           should be written with
-          <a href="https://en.wikipedia.org/wiki/ASCII">ASCII</a>
+          <a href="https://en.wikipedia.org/wiki/UTF-16">UTF-16</a>
```

Comment:
Correct me if I am wrong, ASCII isn't a subset of UTF16? It is completely different?
https://javarevisited.blogspot.com/2015/02/difference-between-utf-8-utf-16-and-utf.html
> UTF-8 is compatible with ASCII while UTF-16 is incompatible with ASCII

If so then we have to say both.

Reply:
Yup, I knew that ASCII was a subset of UTF8, and falsely assumed that ASCII was a subset of UTF16 by transitivity. I will update this.

Reply:
Going based on the title, I assume UTF 8 is also supported?

Maybe we are getting too technical here? The main thing is we support natural Unicode. The exact file format depends more on what is reading in the file, than the grammar itself, right? If its us reading the file, then its tied to something outside the grammar and can be looked at to be fixed. If its ANTLR reading the file, then maybe we should just say we are bound by ANTLR's reading capabilities.

Reply:
Ok, after doing more research:

> Going based on the title, I assume UTF 8 is also supported?

Yes.

> Maybe we are getting too technical here?

Probably not; I think it is good to be very specific here. From https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-JavaLetterOrDigit:

> Letters and digits may be drawn from the entire Unicode character set[...]. This allows programmers to use identifiers in their programs that are written in their native languages.

I think this should be our goal, as well; our documentation should reflect that we support all unicode characters now.

We could add support for unicode escapes. We could do this in a similar fashion that Java itself does by first translating the unicode escapes to UTF-16, then passing the CharStream to ANTLR. We have a few issues for unicode escape support, but all have been opened by me from examples found in openjdk; I don't think anyone is actually writing code like this:

```java

public class SupplementaryJavaID1 {
    public static void main(String[] s) {
        int \ud801\udc00abc = 1;
        int \ud802\udc00abc = 2;
        int \ud801\udc01abc = 3;
        int def\ud801\udc00 = 4;
        int \ud801\udc00\ud834\udd7b = 5;

        if (\ud801\udc00abc != 1 ||
            \ud802\udc00abc != 2 ||
            \ud801\udc01abc != 3 ||
            def\ud801\udc00 != 4 ||
            \ud801\udc00\ud834\udd7b != 5) {
                throw new RuntimeException("test failed");
        }
    }



```

Reply:
Also: ASCII is a subset of unicode, so we should just state that we support unicode characters, and not escapes. Done.